### PR TITLE
[Feature]: Support LDAP Authentication for Dashboard Login

### LINKED ISSUE #4008: [Feature]: Support LDAP Authentication for Dashboard Login
### Description

Support LDAP Authentication for Dashboard Login

### Use case/motivation

The current Dashboard login uses an admin-user with a plaintext password(admin-password), which is highly insecure. To enhance security, centralize user management, and align with industry standard practices, we need to support LDAP authentication.

### Describe the solution

_No response_

### Subtasks

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by davedwwang**
## Why are the changes needed?
The current Dashboard login uses an admin-user with a plaintext password, which is highly insecure. To enhance security, centralize user management, and align with industry standard practices, we need to switch to LDAP authentication.

Close #4008.

## Brief change log
- Introduced three new login-related configurations in AmoroManagementConf.
- Added LdapPasswdAuthenticationProvider to support LDAP integration (implements PasswdAuthenticationProvider).
- Updated LoginController to use PasswdAuthenticationProvider for user login validation.

## How was this patch tested?

- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no) yes
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented) not documented

### CODE REVIEW

**Comment by davedwwang** [comment]
hi @turboFei Would you mind reviewing this? I’d really appreciate it. Thanks!

File: amoro-ams/src/main/java/org/apache/amoro/server/dashboard/controller/LoginController.java [hunk 1]
```diff
@@ -49,10 +50,11 @@ public void login(Context ctx) {
...
-    if (adminUser.equals(user) && (adminPassword.equals(pwd))) {
-      ctx.sessionAttribute("user", new SessionInfo(adminUser, System.currentTimeMillis() + ""));
-      ctx.json(OkResponse.of("success"));
-    } else {
+    DefaultPasswordCredential credential = new DefaultPasswordCredential(user, pwd);
+    try {
+      dashboardServer.getLoginAuthProvider().authenticate(credential);
+    } catch (Exception e) {
+      LOG.error("authenticate user {} failed", user, e);
       throw new RuntimeException("invalid user " + user + " or password!");
```
Comment (turboFei) [inline amoro-ams/src/main/java/org/apache/amoro/server/dashboard/controller/LoginController.java]: return the cause message as well

File: amoro-ams/src/test/resources/ldap/users.ldif [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+dn: cn=user1,ou=Users,dc=test,dc=com
+objectClass: inetOrgPerson
+objectClass: organizationalPerson
+objectClass: person
+objectClass: top
+cn: user1
+sn: Ldap
+uid: ldaptest1
+userPassword: 12345
```
Comment (turboFei) [inline amoro-ams/src/test/resources/ldap/users.ldif]: new line at the end

File: amoro-ams/src/main/java/org/apache/amoro/server/dashboard/DashboardServer.java [hunk 3]
```diff
@@ -92,6 +92,8 @@ public class DashboardServer {
 
   private final PasswdAuthenticationProvider basicAuthProvider;
   private final TokenAuthenticationProvider jwtAuthProvider;
+
+  private final PasswdAuthenticationProvider loginAuthProvider;
```
Comment (turboFei) [inline amoro-ams/src/main/java/org/apache/amoro/server/dashboard/DashboardServer.java]: how about moving the loginAuthProvider into LoginController?

File: amoro-ams/src/main/java/org/apache/amoro/server/dashboard/controller/LoginController.java [hunk 4]
```diff
@@ -49,10 +50,11 @@ public void login(Context ctx) {
     Map<String, String> bodyParams = ctx.bodyAsClass(Map.class);
     String user = bodyParams.get("user");
     String pwd = bodyParams.get("password");
-    if (adminUser.equals(user) && (adminPassword.equals(pwd))) {
-      ctx.sessionAttribute("user", new SessionInfo(adminUser, System.currentTimeMillis() + ""));
-      ctx.json(OkResponse.of("success"));
```
Comment (turboFei) [inline amoro-ams/src/main/java/org/apache/amoro/server/dashboard/controller/LoginController.java:54]: Seems this part of logic is missed after `dashboardServer.getLoginAuthProvider().authenticate(credential);`

File: amoro-ams/src/main/java/org/apache/amoro/server/authentication/LdapPasswdAuthenticationProvider.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+public class LdapPasswdAuthenticationProvider implements PasswdAuthenticationProvider {
+  public static final Logger LOG = LoggerFactory.getLogger(LdapPasswdAuthenticationProvider.class);
+
+  private String ldapUrl;
+  private String ldapUserParttern;
+  private MessageFormat formatter;
+
+  public LdapPasswdAuthenticationProvider(Configurations conf) {
+    this.ldapUrl = conf.get(AmoroManagementConf.HTTP_SERVER_LOGIN_AUTH_LDAP_URL);
+    this.ldapUserParttern = conf.get(AmoroManagementConf.HTTP_SERVER_LOGIN_AUTH_LDAP_USERPARTTERN);
```
Comment (turboFei) [inline amoro-ams/src/main/java/org/apache/amoro/server/authentication/LdapPasswdAuthenticationProvider.java]: check the ldapUrl and ldapUserParttern non empty

File: amoro-ams/src/main/java/org/apache/amoro/server/dashboard/controller/LoginController.java [hunk 6]
```diff
@@ -49,10 +50,11 @@ public void login(Context ctx) {
     Map<String, String> bodyParams = ctx.bodyAsClass(Map.class);
     String user = bodyParams.get("user");
     String pwd = bodyParams.get("password");
```
Comment (turboFei) [inline amoro-ams/src/main/java/org/apache/amoro/server/dashboard/controller/LoginController.java:58]: check the user and pwd non empty

File: amoro-ams/src/main/java/org/apache/amoro/server/dashboard/DashboardServer.java [hunk 7]
```diff
@@ -92,6 +92,7 @@ public class DashboardServer {
 
   private final PasswdAuthenticationProvider basicAuthProvider;
   private final TokenAuthenticationProvider jwtAuthProvider;
+
```
Comment (turboFei) [inline amoro-ams/src/main/java/org/apache/amoro/server/dashboard/DashboardServer.java]: nit: unnecessary change

File: amoro-ams/src/main/java/org/apache/amoro/server/AmoroManagementConf.java [hunk 8]
```diff
@@ -328,6 +328,29 @@ public class AmoroManagementConf {
...
 
+  public static final ConfigOption<String> HTTP_SERVER_LOGIN_AUTH_PROVIDER =
+      ConfigOptions.key("http-server.login-auth-provider")
+          .stringType()
+          .defaultValue(DefaultPasswdAuthenticationProvider.class.getName())
+          .withDescription(
+              "User-defined login authentication implementation of"
+                  + " org.apache.amoro.authentication.PasswdAuthenticationProvider");
+
+  public static final ConfigOption<String> HTTP_SERVER_LOGIN_AUTH_LDAP_USERPARTTERN =
```
Comment (turboFei) [inline amoro-ams/src/main/java/org/apache/amoro/server/AmoroManagementConf.java]: typo:   `PARTTERN` -> `PATTERN`

Comment (turboFei) [reply amoro-ams/src/main/java/org/apache/amoro/server/AmoroManagementConf.java]: recommend: `USER_PATTERN`.

**Review by turboFei — APPROVED** [review]
LGTM overall, thanks

File: amoro-ams/src/main/java/org/apache/amoro/server/AmoroManagementConf.java (hunk 8 above)
Comment (davedwwang) [reply amoro-ams/src/main/java/org/apache/amoro/server/AmoroManagementConf.java]: done

File: amoro-ams/src/main/java/org/apache/amoro/server/dashboard/DashboardServer.java (hunk 7 above)
Comment (davedwwang) [reply amoro-ams/src/main/java/org/apache/amoro/server/dashboard/DashboardServer.java]: done

**Comment by zhoujinsong** [comment]
Thanks for the contribution! @davedwwang 
Thanks for the review! @turboFei
